opm/simulators/aquifers/AquiferConstantFlux.hpp

```diff
@@ -58,7 +58,8 @@ class AquiferConstantFlux : public AquiferInterface<TypeTag>
         , aquifer_data_            (aquifer)
         , connection_flux_         (connections_.size(), Eval{0})
     {
-        this->initializeConnections();
+        const auto connected_face_area = this->initializeConnections();
+        this->area_fraction_ = this->computeFaceAreaFraction(connected_face_area);
```

Comment:
At least in one build, the `area_fraction_` caused problem for the parallel restarting. But I have not got clear understanding of the issue. Will post more later.

Reply:
I need to check whether it is my building is problematic since the output is not very reasonable.

Reply:
> I need to check whether it is my building is problematic

Yes please.  For what it's worth, we need **something** like the area fraction for parallel restart if the set of aquifer connections happens to be split among multiple processes.

> since the output is not very reasonable.

What kind of symptoms do you see?

Reply:
Using the following code to output inside function `computeFaceAreaFraction`, 
```c++
        const double fraction =(tot_face_area > 0.0)
            ? connected_face_area / tot_face_area
            : 0.0;

        const auto rank = this->ebos_simulator_.vanguard()
                .grid().comm().rank();
        std::cout << " id " << this->aquiferID() << " rank " << rank << " connected_face_area " << connected_face_area << " total_face_area " << tot_face_area  << " fraction " << fraction << std::endl;
```


For running with 1 process `computeFaceAreaFraction`,  (the first number is line number, you can ignore it.)
```
1234  id 2 rank 0 connected_face_area 25000 total_face_area 25000 fraction 1
1235  id 1 rank 0 connected_face_area 25000 total_face_area 25000 fraction 1
```

For running with 5 processes
```
 1252  id 2 rank 0 connected_face_area 20000 total_face_area 45000 fraction 0.444444
 1253  id 1 rank 4 connected_face_area 0 total_face_area 45000 fraction 0
 1254  id 1 rank 1 connected_face_area 0 total_face_area 45000 fraction 0
 1255  id 1 rank 2 connected_face_area 0 total_face_area 45000 fraction 0
 1256  id 1 rank 3 connected_face_area 25000 total_face_area 45000 fraction 0.555556
 1257  id 2 rank 2 connected_face_area 0 total_face_area 5000 fraction 0
 1258  id 1 rank 0 connected_face_area 0 total_face_area 5000 fraction 0
 1259  id 2 rank 3 connected_face_area 5000 total_face_area 5000 fraction 1
 1260  id 2 rank 4 connected_face_area 0 total_face_area 5000 fraction 0
 1261  id 2 rank 1 connected_face_area 0 total_face_area 5000 fraction 0
```

It is a build with clang, so I need to double check with other build. I will send you the case through slack.